backend/main/src/Shared/Adapters/HonoAdapter.ts

```diff
@@ -49,18 +49,35 @@ export const honoOkResponseAdapter = (
 
 /**
  * This is the just wrapper of the Hono.
- * This returns 500 Bad Request.
- * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/500
+ * This returns 201 OK.
+ * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/201
  * @param c
- * @param errorMessage - the error message to return.
+ * @param data
  */
-export const honoInternalServerErrorAdapter = (
+export const honoCreatedResponseAdapter = (
 	c: Context,
-	errorMessage: string,
+	data: object | string | null,
 ): Response => {
+	if (typeof data === "string") {
+		c.header("Content-Type", "text/json");
+		c.status(201);
+		return c.text(data);
+	}
+	c.header("Content-Type", "application/json");
+	c.status(201);
+	return c.json(JSON.stringify(data));
+};
+
+/**
+ * This is the just wrapper of the Hono.
+ * This returns 404 Not Found.
+ * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404
+ * @param c
+ */
+export const honoMethodNotFoundAdapter = (c: Context): Response => {
 	c.header("Content-Type", "text/plain");
-	c.status(500);
-	return c.text(errorMessage);
+	c.status(405);
+	return c.text("Method Not Allowed");
```

Comment:
Can I ask why you added this method instead of using `honoMethodNotAllowAdapter`?

Reply:
Sorry, this is my mistake. What I wanted to add is a `Not Found` response.